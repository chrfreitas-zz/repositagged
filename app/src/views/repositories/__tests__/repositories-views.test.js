import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import Repositories from '../index';

describe('Repositories View', () => {
  const props = {
    repositories: [],
    match: {
      params: {
        username: 'xyz',
      },
    },
    getRepositories: jest.fn(),
  };

  it('should match with snapshot', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Repositories {...props} />);
    expect(result).toMatchSnapshot();
  });

  describe('when it did mount', () => {
    it('should call getRepositories once', () => {
      shallow(<Repositories {...props} />);
      expect(props.getRepositories).toHaveBeenCalledTimes(1);
    });
  });
});
