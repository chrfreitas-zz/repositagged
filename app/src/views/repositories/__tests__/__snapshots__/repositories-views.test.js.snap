// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories view should match with snapshot 1`] = `
<React.Fragment>
  <div
    className="repositories"
  >
    <div
      className="repositories__row"
    >
      <Input
        className="input input--medium"
        placeholder="search by tag"
      />
    </div>
    <div
      className="repositories__row"
    >
      <Table
        body={
          Array [
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s",
              "id": 1,
              "language": "go",
              "name": "kubernetes",
              "tag": "",
            },
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s",
              "id": 2,
              "language": "python",
              "name": "django",
              "tag": "",
            },
            Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Laboriosam, nihil.s",
              "id": 3,
              "language": "javascript",
              "name": "jest",
              "tag": "",
            },
          ]
        }
        header={
          Array [
            "Repository",
            "Description",
            "Language",
            "Tags",
            "",
          ]
        }
      />
    </div>
  </div>
</React.Fragment>
`;
